qiskit/chemistry/ground_state_calculation/adapt_vqe.py

```diff
@@ -0,0 +1,270 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+A ground state calculation employing the AdaptVQE algorithm.
+"""
+
+from typing import Optional, List
+import logging
+import re
+import warnings
+import numpy as np
+
+from qiskit.aqua import AquaError
+from qiskit.aqua.algorithms import VQEResult
+from qiskit.aqua.operators import LegacyBaseOperator, WeightedPauliOperator
+from qiskit.aqua.utils.validation import validate_min
+from qiskit.chemistry.components.variational_forms import UCCSD
+from qiskit.chemistry.core import MolecularGroundStateResult
+from qiskit.chemistry.drivers import BaseDriver
+from qiskit.chemistry.qubit_transformations import FermionicTransformation
+from .ground_state_calculation import GroundStateCalculation
+from .mes_factory import MESFactory
+
+logger = logging.getLogger(__name__)
+
+
+class AdaptVQE(GroundStateCalculation):
+    """A ground state calculation employing the AdaptVQE algorithm."""
+
+    def __init__(self,
+                 transformation: FermionicTransformation,
+                 solver: MESFactory,
+                 threshold: float = 1e-5,
+                 delta: float = 1,
+                 max_iterations: Optional[int] = None,
+                 ) -> None:
+        """
+        Args:
+            transformation: a fermionic driver to operator transformation strategy.
+            solver: a minimum eigensolver factory which uses the UCCSD variational form.
+            threshold: the energy convergence threshold. It has a minimum value of 1e-15.
+            delta: the finite difference step size for the gradient computation. It has a minimum
+                   value of 1e-5.
+            max_iterations: the maximum number of iterations of the AdaptVQE algorithm.
+        """
+        validate_min('threshold', threshold, 1e-15)
+        validate_min('delta', delta, 1e-5)
+
+        super().__init__(transformation)
+
+        self._solver = solver
+        self._threshold = threshold
+        self._delta = delta
+        self._max_iterations = max_iterations
+
+    def returns_groundstate(self) -> bool:
+        return False
```

Comment:
This should return `True` if it simply indicates that the algorithm supports `aux_operators`.